geo/src/algorithm/area.rs

```diff
@@ -27,12 +35,25 @@ where
     // of the coordinates, but it is not fool-proof to
     // divide by the length of the linestring (eg. a long
     // line-string with T = u8)
-    let shift = linestring.0[0];
+    let shift = linestring.coord(0).unwrap();
 
     let mut tmp = T::zero();
-    for line in linestring.lines() {
-        use crate::MapCoords;
-        let line = line.map_coords(|c| c - shift);
+    for i in 0..linestring.num_coords() - 1 {
+        let mut c1 = Coord {
+            x: linestring.coord(i).unwrap().x(),
+            y: linestring.coord(i).unwrap().y(),
+        };
+        c1.x = c1.x - shift.x();
```

Comment:
I switched these to not use `-=` to avoid having to require the `SubAssign` type on `T`. Maybe requiring `SubAssign` and similar is such a low bar that we should require it?